### PR TITLE
Chore: Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by jstoffan**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #613

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        ],
+    ],
+    env: {
+        dev: {
+            plugins: ['flow-react-proptypes'],
+        },
+        npm: {
+            plugins: [
+                [
+                    'babel-plugin-transform-require-ignore',
```
Comment (priyajeet) [inline babel.config.js]: I think we should be able to remove this one completely from npm.
So that babel output has SCSS imports, which makes it unnecessary to import the CSS file.
(like BRUI)

But that can be another commit.

File: src/components/makeResponsive.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ import {
     CLASS_IS_MEDIUM,
 } from '../constants';
 
-type Props = {
+type PropsShape = {
```
Comment (priyajeet) [inline src/components/makeResponsive.js]: Whats this?

Comment (jstoffan) [reply src/components/makeResponsive.js]: The class properties transform throws a fit unless the function is typed with `Props: { whatever }`. This was the simplest way to keep the function typed without throwing errors on compile. I can change the name if you have an alternative suggestion.

File: babel.config.js (hunk 1 above)
Comment (jstoffan) [reply babel.config.js]: I'd like to do that separately, since I believe it would be a breaking change. It would probably lead to duplicate stylesheets showing up for users who import the components, as they may already have the compiled stylesheet imported at the top level of their project.
